camkes/ast/liftedast.py

```diff
@@ -59,7 +59,7 @@ def assembly(self):
 
     def filter(self, function=None):
         if function is None:
-            function = lambda x: x is not None
+            def function(x): return x is not None
```

Comment:
In the old code `function` was a variable holding a function or `None`. In your change a local function named `function` is introduced that shadows the actual variable. I'm not sure how much difference this makes in Python in the end, but the cleaner way would be be:
```
def my_function(x): return x is not None
function = my_function
```
The original code was basically a shorter version of this, using `lambda` to create an anonymous function. Not sure what "ref E731" in the commit message is supposed to mean, I could not relate this to anything. Also, while some time back there have been attempts to deprecate `lambda`, it well accepted now, see https://mail.python.org/pipermail/python-dev/2006-February/060415.html

Reply:
ah ok, found it at https://www.flake8rules.com/rules/E731.html
But still, we should not shadow the variable here.

Reply:
Yes. It's a flake 8 rule.

> In the old code `function` was a variable holding a function or `None`. In your change a local function named `function` is introduced that shadows the actual variable. I'm not sure how much difference this makes in Python in the end, but the cleaner way would be be:
> 
> ```
> def my_function(x): return x is not None
> function = my_function
> ```
> 
> The original code was basically a shorter version of this, using `lambda` to create an anonymous function. Not sure what "ref E731" in the commit message is supposed to mean, I could not relate this to anything. Also, while some time back there have been attempts to deprecate `lambda`, it well accepted now, see https://mail.python.org/pipermail/python-dev/2006-February/060415.html

Let me change it.